async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -53,19 +52,29 @@ impl Consumer<Config> {
     ///     ..Default::default()
     /// }).await?;
     ///
-    /// let mut messages = consumer.stream()?.take(100);
+    /// let mut messages = consumer.stream().await?.take(100);
     /// while let Some(Ok(message)) = messages.next().await {
     ///   println!("got message {:?}", message);
     ///   message.ack().await?;
     /// }
     /// Ok(())
     /// # }
     /// ```
-    pub fn stream(&self) -> Result<Stream, Error> {
-        let sequence = self.sequence(10)?;
-        let try_flatten = sequence.try_flatten();
-
-        Ok(try_flatten)
+    pub async fn stream(&self) -> Result<Stream<'_>, Error> {
+        Stream::stream(
+            BatchConfig {
+                batch: 100,
+                expires: Some(Duration::from_secs(30).as_nanos().try_into().unwrap()),
+                no_wait: false,
+                max_bytes: 0,
+                idle_heartbeat: Duration::default(),
+            },
+            self,
+        )
+        .await
+    }
+    pub async fn stream_with_config(&self, config: BatchConfig) -> Result<Stream<'_>, Error> {
```

Comment:
Since we know we can easily run into issues with this, maybe we wanna hold off on the with_config variant?

```suggestion
    pub async fn stream_with_config(&self, config: BatchConfig) -> Result<Stream<'_>, Error> {
```

Reply:
@caspervonb  I will follow up today with:

1. make this private
2. expose only batch, max bytes and hearbets.

It's not released yet, so I would not bother with removing it, as it requires removing tests that will be needed the same day (pull with heartbeat).